tests/geobase/checks_FairGeoBasicShape.h

```diff
@@ -0,0 +1,167 @@
+/********************************************************************************
+ *   Copyright (C) 2024 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH     *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+#include "FairGeoBasicShape.h"
+#include "FairGeoRotation.h"
+#include "FairGeoTransform.h"
+#include "FairGeoVector.h"
+#include "FairGeoVolume.h"
+#include "generate_TestInput.h"
+
+#include <TArrayD.h>
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+#include <cstdio>
+#include <string>
+
+using Catch::Matchers::WithinRel;
+
+#ifndef TESTS_GEOBASE_CHECKS_FAIRGEOBASICSHAPE_H
+#define TESTS_GEOBASE_CHECKS_FAIRGEOBASICSHAPE_H
+
+namespace fairroot::tests
+{
+
+auto check_FairGeoTransform(FairGeoTransform& _trans)
+{
+    const FairGeoRotation& rot = _trans.getRotMatrix();
+    CHECK(rot.isUnitMatrix());
+
+    const FairGeoVector& vec = _trans.getTransVector();
+    CHECK_THAT(vec.getValues(0), WithinRel(0., 0.001));
+    CHECK_THAT(vec.getValues(1), WithinRel(0., 0.001));
+    CHECK_THAT(vec.getValues(2), WithinRel(0., 0.001));
+
+    const FairGeoVector& vec_cm = _trans.getTranslation();
+    CHECK_THAT(vec_cm.getValues(0), WithinRel(0., 0.001));
+    CHECK_THAT(vec_cm.getValues(1), WithinRel(0., 0.001));
+    CHECK_THAT(vec_cm.getValues(2), WithinRel(0., 0.001));
+}
+
+auto check_FairGeoShape_Construction(FairGeoBasicShape& shape,
+                                     std::string name,
+                                     int numPoints,
+                                     int numParams,
+                                     TArrayD* _param)
+{
```

Comment:
I think you only read from `_param`? Can you turn it into a `const TArrayD*`?

`numParams` and `_param->GetSize()` seem to need to match?

1. ```suggestion
   auto check_FairGeoShape_Construction(FairGeoBasicShape& shape,
                                        std::string name,
                                        int numPoints,
                                        TArrayD* _param)
   {
       int numParams = (_param ? _param->GetSize() : 0);
   ```

2. ```suggestion
   auto check_FairGeoShape_Construction(FairGeoBasicShape& shape,
                                        std::string name,
                                        int numPoints,
                                        int numParams,
                                        TArrayD* _param)
   {
       REQUIRE(numParams == (_param ? _param->GetSize() : 0));
   ```

Reply:
Done.